crates/gen/src/types/function.rs

```diff
@@ -51,37 +51,46 @@ impl Function {
             link = "onecoreuap";
         }
 
-        if cfg!(windows) {
+        let body = if cfg!(windows) {
             if signature.has_query_interface() {
                 let leading_params = &signature.params[..signature.params.len() - 2];
-                let params = signature.gen_win32_params(leading_params, gen);
                 let args = leading_params.iter().map(|p| p.gen_win32_abi_arg());
 
                 quote! {
-                    pub unsafe fn #name<#constraints T: ::windows::Interface>(#params) -> ::windows::Result<T> {
-                        #[link(name = #link)]
-                        extern "system" {
-                            pub fn #name(#(#abi_params),*) #abi_return_type;
-                        }
-                        let mut result__ = ::std::option::Option::None;
-                        #name(#(#args,)* &<T as ::windows::Interface>::IID, ::windows::Abi::set_abi(&mut result__)).and_some(result__)
+                    #[link(name = #link)]
+                    extern "system" {
+                        pub fn #name(#(#abi_params),*) #abi_return_type;
                     }
+                    let mut result__ = ::std::option::Option::None;
+                    #name(#(#args,)* &<T as ::windows::Interface>::IID, ::windows::Abi::set_abi(&mut result__)).and_some(result__)
                 }
             } else {
                 quote! {
-                    pub unsafe fn #name<#constraints>(#params) #return_type {
-                        #[link(name = #link)]
-                        extern "system" {
-                            pub fn #name(#(#abi_params),*) #abi_return_type;
-                        }
-                        #name(#(#args),*)
+                    #[link(name = #link)]
+                    extern "system" {
+                        pub fn #name(#(#abi_params),*) #abi_return_type;
                     }
+                    #name(#(#args),*)
+                }
+            }
+        } else {
+            quote! {
+                panic!("Unsupported target OS");
```

Comment:
nit: I wonder if the `unimplemented!` is a slightly better choice rather than `panic`. It's the same thing, but it's more clear in code and adds "unimplemented" to the panic message.

Reply:
Check, I agree :)